src/main/java/organice/logic/parser/MatchCommandParser.java

```diff
@@ -0,0 +1,38 @@
+package organice.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+import static organice.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static organice.logic.parser.CliSyntax.PREFIX_NRIC;
+
+import organice.logic.commands.MatchCommand;
+import organice.logic.parser.exceptions.ParseException;
+import organice.model.person.Nric;
+
+/**
+ * Parses input arguments and creates a new MatchCommand object
+ */
+public class MatchCommandParser implements Parser<MatchCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the EditCommand
+     * and returns an EditCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public MatchCommand parse(String args) throws ParseException {
+        requireNonNull(args);
+        //TODO: Add prefix for all/
+        ArgumentMultimap argMultimap = ArgumentTokenizer.tokenize(args, PREFIX_NRIC);
+        String output;
```

Comment:
if u use `ic/all` it will make ur life easier i believe

Reply:
To clarify, this PR is meant to implement `match ic/**NRIC**` only. The purpose of separating into two PRs is to get feedback for my initial implementation and to avoid overwhelming the reviewers.